tests/experimental/wgit/test_signal_sparsity.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+
+import pytest
+import torch
+
+from fairscale.experimental.wgit.signal_sparsity import SignalSparsity, _get_k_for_topk, _top_k_total_size
+
+TRIALS = 5
+TENSOR_DIM_LOWER = 40
+TENSOR_DIM_UPPER = 100
+DIM_LIST = [None, 0, 1]
+SPARSITY_ATOL = 8e-2
+FFT_RECONS_ATOL = 2e-3
+
+
+def test_validate_conf():
+    """Tests the config validation for the Signal Sparsity class."""
+
+    def kwargs(vals_list):
+        """Maps the values in input vals_list to the keys in arg_key_list"""
+
+        arg_key_list = [
+            "sst_top_k_element",
+            "sst_top_k_percent",
+            "sst_top_k_dim",
+            "dst_top_k_element",
+            "dst_top_k_percent",
+            "dst_top_k_dim",
+        ]
+        return dict(zip(arg_key_list, vals_list))
+
+    # Validate assertion error is raised when, either
+    # 1. both of sst (or dst) percent and element is set to values (not None).
+    # 2. both of sst (or dst) percent and element is set to None.
+    element = random.randint(0, TENSOR_DIM_UPPER)
+    percent = random.uniform(0, 100)
+    dim = random.randint(-1, 3)
+
+    pytest.raises(AssertionError, SignalSparsity, **kwargs([element, percent, dim, element, None, dim]))
+    pytest.raises(AssertionError, SignalSparsity, **kwargs([element, None, dim, element, percent, dim]))
+    pytest.raises(AssertionError, SignalSparsity, **kwargs([None, None, dim, element, None, dim]))
+    pytest.raises(AssertionError, SignalSparsity, **kwargs([element, None, dim, None, None, dim]))
+
+
+def test_sst_sparsity_level():
+    """Tests the dense_to_sst method of SignalSparsity in terms of generating SST at the correct
+    sparsity level.
+    """
+
+    def get_sst(sst_top_k_percent, dst_top_k_percent=100):
+        sparser_2d = SignalSparsity(sst_top_k_percent=sst_top_k_percent, dst_top_k_percent=100)
```

Comment:
=100 here or should be =dst_top_k_percent?

Reply:
This test was for sst only, so `dst_top_k_percent=100` (no change as choosing top 100% or all). When we  do add dst implementation didn't want that to impact sst only tests.

Reply:
I see. why dst_top_k_percent default to 100 on line 55? Shouldn't it like sst_top_k_percent that don't have a default?